edward/inferences/renyi_divergence.py

```diff
@@ -0,0 +1,191 @@
+                        if isinstance(qx, RandomVariable):
+                            qx_copy = copy(qx, scope=scope)
+                            dict_swap[x] = qx_copy.value()
+                        else:
+                            dict_swap[x] = qx
+
+                for z, qz in six.iteritems(self.latent_vars):
+                    # Copy q(z) to obtain new set of posterior samples.
+                    qz_copy = copy(qz, scope=scope)
+                    dict_swap[z] = qz_copy.value()
+                    q_log_prob[s] += tf.reduce_sum(
+                        self.scale.get(z, 1.0)
+                        * qz_copy.log_prob(dict_swap[z]))
+
+                for z in six.iterkeys(self.latent_vars):
+                    z_copy = copy(z, dict_swap, scope=scope)
+                    p_log_prob[s] += tf.reduce_sum(
+                        self.scale.get(z, 1.0)
+                        * z_copy.log_prob(dict_swap[z]))
+
+                for x in six.iterkeys(self.data):
+                    if isinstance(x, RandomVariable):
+                        x_copy = copy(x, dict_swap, scope=scope)
+                        p_log_prob[s] += tf.reduce_sum(
+                            self.scale.get(x, 1.0)
+                            * x_copy.log_prob(dict_swap[x]))
+
+            logF = [p - q for p, q in zip(p_log_prob, q_log_prob)]
+
+            if self.backward_pass == 'max':
+                logF = tf.stack(logF)
+                logF = tf.reduce_max(logF, 0)
+                loss = tf.reduce_mean(logF)
+            elif self.backward_pass == 'min':
+                logF = tf.stack(logF)
+                logF = tf.reduce_min(logF, 0)
+                loss = tf.reduce_mean(logF)
+            elif isclose(self.alpha, 1.0, abs_tol=10e-3):
+                loss = tf.reduce_mean(logF)
+            else:
+                logF = tf.stack(logF)
+                logF = logF * (1 - self.alpha)
+                logF_max = tf.reduce_max(logF, 0)
+                logF = tf.log(
+                    tf.maximum(1e-9,
+                               tf.reduce_mean(tf.exp(logF - logF_max), 0)))
+                logF = (logF + logF_max) / (1 - self.alpha)
+                loss = tf.reduce_mean(logF)
+            loss = -loss
+
+            if self.logging:
+                p_log_prob = tf.reduce_mean(p_log_prob)
+                q_log_prob = tf.reduce_mean(q_log_prob)
+                tf.summary.scalar("loss/p_log_prob", p_log_prob,
+                                  collections=[self._summary_key])
+                tf.summary.scalar("loss/q_log_prob", q_log_prob,
+                                  collections=[self._summary_key])
+
+            grads = tf.gradients(loss, var_list)
+            grads_and_vars = list(zip(grads, var_list))
+            return loss, grads_and_vars
+        else:
+            raise NotImplementedError(
+                "Variational Renyi inference only works with reparameterizable"
+                " models")
+
+#########
```

Comment:
This function is only used in one location and is a one-liner; could you write that line instead of defining a new function?